app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_user
+  
+  def require_user
+    render file: "/public/404" if current_user.nil? || current_admin? == false
+  end
```

Comment:
I think this can just be `require_admin` from the dashboard controller

Reply:
Agreed. I changed it in my current branch (`jww_user_story_52`), but I'll update it here as well.